hadoop-hdds/container-service/src/main/java/org/apache/hadoop/ozone/container/keyvalue/helpers/KeyValueContainerUtil.java

```diff
@@ -254,6 +255,13 @@ public static void parseKVContainerData(KeyValueContainerData kvContainerData,
       if (!isBlockMetadataSet) {
         initializeUsedBytesAndBlockCount(store, kvContainerData);
       }
+
+      // If the container is missing a chunks directory, possibly due to the
+      // bug fixed by HDDS-6235, create it here.
+      File chunksDir = new File(kvContainerData.getChunksPath());
+      if (!chunksDir.exists()) {
```

Comment:
Hi ethan, as I read, the chunks directory should be created in the TarContainerPacker while unpacking, then is there a case that we have to create this directory again in parseKVContainerData?

Reply:
Hi @guihecheng, thanks for checking out the PR! There's two cases we want to fix here:
1. Empty containers that may need to be replicated in the future.
    - ReplicationManager is supposed to delete these, so this is more of a defensive code path.
2. Existing empty containers that got replicated before this patch and HDDS-5359 and lost their chunks directory.

Most of this patch is aimed at the first case, but these lines are aimed at the second case. We want to repair the existing containers that already lost their chunks directory so they do not get marked unhealthy by the scrubber (if running). Also, a container missing its chunks directory will fail replication on the datanode even before this patch, so SCM constantly reports it as under-replicated. There may be a lot of containers in this state due to the issue reported in HDDS-5359, which prevented many empty containers from being deleted.

Reply:
parseKVContainerData is called on container import and startup. We are targeting the startup run to fix all the containers after the patch is applied. When this is called on import it should be a no-op since the chunks dir is now traveling with the tar as you mentioned.

Reply:
Oh, thanks for the explanation, actually we've seen similar container replicas with no "chunks" directory in our cluster, and I think this patch fixes the problem indeed.
LGTM+1